crates/matrix-sdk-ui/src/timeline/pinned_events_loader.rs

```diff
@@ -1,31 +1,27 @@
-use std::{collections::BTreeMap, fmt::Formatter, sync::Arc};
+use std::{fmt::Formatter, sync::Arc};
 
 use itertools::Itertools;
-use matrix_sdk::{event_cache::paginator::PaginatorError, Room, SendOutsideWasm, SyncOutsideWasm};
+use matrix_sdk::{
+    event_cache::paginator::PaginatorError, pinned_events_cache::PinnedEventCache, Room,
+    SendOutsideWasm, SyncOutsideWasm,
+};
 use matrix_sdk_base::deserialized_responses::SyncTimelineEvent;
 use ruma::{EventId, MilliSecondsSinceUnixEpoch, OwnedEventId};
 use thiserror::Error;
-use tokio::sync::{RwLock, Semaphore};
+use tokio::sync::Semaphore;
+use tracing::info;
 
 /// Utility to load the pinned events in a room.
 pub struct PinnedEventsLoader {
     room: Arc<Box<dyn PinnedEventsRoom>>,
     max_events_to_load: usize,
     max_concurrent_requests: usize,
-    cache: PinnedEventCache,
 }
 
 impl PinnedEventsLoader {
     /// Creates a new `PinnedEventsLoader` instance.
     pub fn new(room: Box<dyn PinnedEventsRoom>, max_events_to_load: usize) -> Self {
-        Self {
-            room: Arc::new(room),
-            max_events_to_load,
-            max_concurrent_requests: 10,
-            cache: PinnedEventCache {
-                inner: Arc::new(InnerPinnedEventCache { events: Default::default() }),
-            },
-        }
+        Self { room: Arc::new(room), max_events_to_load, max_concurrent_requests: 10 }
```

Comment:
Probably best we put these values in separate`const`s

Reply:
https://github.com/matrix-org/matrix-rust-sdk/pull/3796/commits/60a86d7eb4970d84f1d9e0b994c68ecaaa2e9644

Reply:
`max_events_to_load` is an argument so clients can provide the max value themselves instead of being limited by the SDK.